frontend/src/components/Header.js

```diff
@@ -0,0 +1,59 @@
+import React, { Component, PropTypes } from 'react'
+
+class Header extends Component {
+  render = () => {
+    const signedIn = this.props.signedIn
+    const section = this.props.section
+    var activeClass = (title) => {
+      var forClass = "exploreMapsButton"
+      forClass += " " + title + "Maps"
+      if (title == "my" && section == "mine" ||
+          title == section) forClass += " active"
+      return forClass
+    }
+
+    const explore = section == "mine" || section == "active" || section == "shared" || section == "featured"
+    const mapper = section == "mapper"
+    const apps = section == "registered" || section == "authorized"
+
+    return (
+      <div className="exploreMapsBar exploreElement">
+        <div className="exploreMapsMenu">
+          <div className="exploreMapsCenter">
+            {signedIn && explore ? <a href="/explore/mine" className={activeClass("my")}>
+              <div className="exploreMapsIcon"></div>
+              My Maps
+            </a> : null }
+            {signedIn && explore ? <a href="/explore/shared" className={activeClass("shared")}>
+              <div className="exploreMapsIcon"></div>
+              Shared With Me
+            </a> : null }
+            {explore ? <a href={signedIn ? "/" : "/explore/active"}  className={activeClass("active")}>
+              <div className="exploreMapsIcon"></div>
+              Recently Active
+            </a> : null }
+            {!signedIn && explore ? <a href="/explore/featured" className={activeClass("featured")}>
+              <div className="exploreMapsIcon"></div>
+              Featured Maps
+            </a> : null }
+            {mapper ? <div className='exploreMapsButton active mapperButton'>
+                <img className='exploreMapperImage' width='24' height='24' src={this.props.userAvatar} />
+                <div className='exploreMapperName'>{this.props.userName}&rsquo;s Maps</div>
+                <div className='clearfloat'></div>
+              </div> : null }
+            {apps ? <a href="/oauth/applications" className={"activeMaps exploreMapsButton"  + (section == "registered" ? " active" : "")} data-bypass="true">
+              <div className="exploreMapsIcon"></div>
+              Registered Apps
+            </a> : null }
+            {apps ? <a href="/oauth/authorized_applications" className={"featuredMaps exploreMapsButton" + (section == "authorized" ? " active" : "")} data-bypass="true">
+              <div className="exploreMapsIcon"></div>
+              Authorized Apps
+            </a> : null }
+          </div>
+        </div>
+      </div>
+    )
+  }
+}
+
```

Comment:
Do you know about var/let/const? It's really non-obvious, but it helps out with React code because when you deal with React state, you always try to avoid mutating it.

`var` behaves like the old javascript var. It's function-scoped, and if you leave it off it's a global var.
`const` creates a constant, but almost all the time in my es6/react code I use const. If it's an object or array, you can still add/delete/push/pop properties, so I use const 90% of the time. `const` is also block-scoped (so inside if/for/while, instead of being all over the function), unlike var, which makes it nicer to work with than var.

`let` is more like a traditional variable - it allows mutations. It's also block-scoped. You mostly only need this for changing numbers/strings, or occasionally if you need to replace the entire contents of an array/object. A new es6 pattern is `for (let i = 0; i < 10; i += 1) {`; you can see let replaces var.

Most ES6 people say just stop using var (as we've done here) in es6 code. The reason is that let and const are a) block-scoped and b) communicate more about the intent of the variable.